### ISSUE TITLE
Open current URL and video hints with external video player

### ISSUE THREAD

**Issue by indeedwatson** [philc/vimium#2916]
I know there's addons such as this: https://addons.mozilla.org/en-US/firefox/addon/send-to-mpv-player/

But none that I can find allow to launch, for example, mpv with a keybind.

Additionally it would be super cool if there could be some hinting mode that only brings up urls from certain sites that are likely to be videos and attempt to open that with mpv.

I realize this might be hard if not impossible with the current web extension limitations, so I'm just mentioning this as a suggestion.

**Comment by Bumbadawg** [comment]
@indeedwatson 
Here's my setup.

In Firefox i yank a link with `yf`, then press `Win+y` which executes:
```bash
# displays a notif for 7s of the video being played
notify-send -t 7000 -i video-television \
"Playing" "`xclip -o -sel clip | xargs youtube-dl -e`"

# plays the link in mpv
mpv --fs `xclip -o -sel clip`
```
NB: to bind keys, i use KDE global shortcut but you could bind it with it with Xkeybinds.

You can reuse this trick to :
* download the video, bound to `Win+Y`
```bash
notify-send -t 7000 -i video-x-generic \
"Downloading video" "`xclip -o -sel clip | xargs youtube-dl -e`"

# downloads video in ~/video/
xclip -o -sel clip | xargs youtube-dl -o "~/video/%(title)s.%(ext)s"
```
* download the audio playlist, bound to `Win+M`
```bash
notify-send -t 7000 -i audio-x-generic \
"Downloading audio" "`xclip -o -sel clip | xargs youtube-dl -e`"

# downloads audio playlist (if available) in ~/audio/
xclip -o -sel clip | xargs youtube-dl --yes-playlist -x -o "~/audio/%(title)s.%(ext)s"
```
NB#2: youtube-dl can download from a ton of websites, it'll save you so much time you'll never use closed-source clickodromes again =)

**Comment by gdh1995** [comment]
Yes. It's impossible because normal extensions are allowed to access nothing other than system clipboard and page content.
